packages/viewer/src/components/scene-tree/scene-tree.tsx

```diff
@@ -238,70 +241,85 @@ export class SceneTree {
   }
 
   @Watch('viewer')
-  public handleViewerChanged(
+  public async handleViewerChanged(
     newViewer: HTMLVertexViewerElement | undefined,
     oldViewer: HTMLVertexViewerElement | undefined
-  ): void {
-    if (newViewer !== oldViewer) {
-      if (oldViewer != null) {
-        this.cleanupController();
-        oldViewer.removeEventListener(
-          'sceneReady',
-          this.handleViewerSceneReady
-        );
-        oldViewer.removeEventListener(
-          'connectionChange',
-          this.handleViewerConnectionStatusChange
-        );
-      }
+  ): Promise<void> {
+    if (oldViewer != null) {
+      this.controller = undefined;
+
+      oldViewer.removeEventListener('sceneReady', this.handleViewerSceneReady);
+      oldViewer.removeEventListener(
+        'connectionChange',
+        this.handleViewerConnectionStatusChange
+      );
+    }
+
+    if (newViewer != null) {
+      newViewer.addEventListener('sceneReady', this.handleViewerSceneReady);
+      newViewer.addEventListener(
+        'connectionChange',
+        this.handleViewerConnectionStatusChange
+      );
 
-      if (newViewer != null) {
-        newViewer.addEventListener('sceneReady', this.handleViewerSceneReady);
-        newViewer.addEventListener(
-          'connectionChange',
-          this.handleViewerConnectionStatusChange
-        );
+      const isSceneReady = await newViewer.isSceneReady();
+      console.log('is scene ready', isSceneReady);
```

Comment:
Can we move this to `console.debug` or remove?

Reply:
Will remove.